### PR TITLE
Pin all our dependencies, including pip, six, ipython and ipdb.

### LINKED ISSUE #6001: Conflicting setuptools versions
`make update_docker` calls https://github.com/mozilla/addons-server/blob/master/Makefile-docker#L96 which installs the latest _setuptools_ version. Later on, we install a very specific version of it in https://github.com/mozilla/addons-server/blob/master/requirements/prod.txt#L439 which causes failures if a new _setuptools_ version has been released but we haven't locked it in the requirements (yet):

(irrelevant log output removed)
```
$ make update_docker
docker-compose exec worker make update_deps

Collecting setuptools
  Using cached setuptools-36.2.2-py2.py3-none-any.whl
Installing collected packages: setuptools
  Found existing installation: setuptools 36.2.0
    Uninstalling setuptools-36.2.0:
      Successfully uninstalled setuptools-36.2.0
Successfully installed setuptools-36.2.2

pip install --no-deps --exists-action=w -r requirements/dev.txt
Collecting setuptools==36.2.0 (from -r requirements/prod.txt (line 439))
  Using cached setuptools-36.2.0-py2.py3-none-any.whl
Installing collected packages: certifi, setuptools
  Found existing installation: certifi 2017.4.17
    Uninstalling certifi-2017.4.17:
      Successfully uninstalled certifi-2017.4.17
  Running setup.py install for certifi ... done
  Found existing installation: setuptools 36.2.2
    Uninstalling setuptools-36.2.2:
      Successfully uninstalled setuptools-36.2.2
  Rolling back uninstall of setuptools

Exception:
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/pip/basecommand.py", line 215, in main
    status = self.run(options, args)
  File "/usr/lib/python2.7/site-packages/pip/commands/install.py", line 342, in run
    prefix=options.prefix_path,
  File "/usr/lib/python2.7/site-packages/pip/req/req_set.py", line 784, in install
    **kwargs
  File "/usr/lib/python2.7/site-packages/pip/req/req_install.py", line 851, in install
    self.move_wheel_files(self.source_dir, root=root, prefix=prefix)
  File "/usr/lib/python2.7/site-packages/pip/req/req_install.py", line 1064, in move_wheel_files
    isolated=self.isolated,
  File "/usr/lib/python2.7/site-packages/pip/wheel.py", line 345, in move_wheel_files
    clobber(source, lib_dir, True)
  File "/usr/lib/python2.7/site-packages/pip/wheel.py", line 305, in clobber
    ', '.join(info_dir))
AssertionError: Multiple .dist-info directories: /usr/lib/python2.7/site-packages/setuptools-36.2.2.dist-info, /usr/lib/python2.7/site-packages/setuptools-36.2.0.dist-info
make: *** [install_python_dependencies] Error 2
make: *** [update_docker] Error 2
```

### PR DESCRIPTION
**Pull request by EnTeQuAk**
This removes all dynamically installed python packages from the
installation pipeline and should result in a cleaner system install.

Fixes #6001

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -71,12 +71,12 @@ commands =
...
 commands =
-    pip install --upgrade pip six setuptools
+    pip install -r requirements/system.txt
     pip install -r requirements/flake8.txt
     make flake8
 
 [testenv:docs]
 commands =
-    pip install --upgrade pip six setuptools
+    pip install -r requirements/system.txt
```
Comment (diox) [inline tox.ini]: With that, we should also be able to clean up deps in `docs.txt` right ? The same way you did in `prod.txt`

File: Dockerfile.deploy [hunk 2]
```diff
@@ -34,9 +34,7 @@ RUN yum install -y                                          \
         uwsgi-2.0.13.1-2.el7                                \
         uwsgi-plugin-python                                 \
     && yum clean all                                        \
-    && curl -sSL https://bootstrap.pypa.io/get-pip.py | python \
-    && pip install six \
-    && pip install setuptools
+    && curl -sSL https://bootstrap.pypa.io/get-pip.py | python
```
Comment (diox) [inline Dockerfile.deploy]: That still makes be nervous... 💦

Comment (EnTeQuAk) [reply Dockerfile.deploy]: Very true… in our non-deploy dockerfile we install pip via `RUN yum install -y python-pip` and later upgrade it explicitly via a requirements file. I think I'm going to change this, our local development approach is TONS more secure. Good catch!

File: tox.ini (hunk 1 above)
Comment (EnTeQuAk) [reply tox.ini]: hell yeah, I'm on it.

**Pull request merged**